avm/res/app/session-pool/main.bicep

```
@@ -0,0 +1,273 @@
+output resourceGroupName string = resourceGroup().name
+@description('The management endpoint of the session pool.')
+output managementEndpoint string = sessionPool.properties.poolManagementEndpoint
+
+@description('The principal ID of the system assigned identity.')
+output systemAssignedMIPrincipalId string = sessionPool.?identity.?principalId ?? ''
+
+@description('Optional. Custom container definition.')
+type sessionContainerType = {
+  @description('Optional. Container start command arguments.')
+  args: string[]?
+
+  @description('Optional. Container start command.')
+  command: string[]?
+
+  @description('Optional. Container environment variables.')
+  env: environmentVarType?
+
+  @description('Required. Container image tag.')
+  image: string
+
+  @description('Required. Custom container name.')
+  name: string
+
+  @description('Required. Container resource requirements.')
+  resources: sessionContainerResourcesType
+}[]?
+
+@description('Optional. Environment variable definition for a container. Only used with custom containers.')
+type environmentVarType = {
+  @description('Required. Environment variable name.')
+  name: string
+
+  @description('Optional. Required if value is not set. Name of the Container App secret from which to pull the environment variable value.')
+  secretRef: string?
+
+  @description('Optional. Required if secretRef is not set. Non-secret environment variable value.')
+  value: string?
+}[]?
+
+@description('Optional. Container resource requirements. Only used with custom containers.')
+type sessionContainerResourcesType = {
+  @description('Required. Required CPU in cores, e.g. 0.5.')
+  cpu: string
+
+  @description('Required. Required memory, e.g. "1.25Gi".')
+  memory: string
+}
+
+@description('Optional. Container registry credentials. Only used with custom containers.')
+type sessionRegistryCredentialsType = {
+  @description('Optional. A Managed Identity to use to authenticate with Azure Container Registry. For user-assigned identities, use the full user-assigned identity Resource ID. For system-assigned identities, use "system".')
+  identity: string?
+
+  @description('Optional. The name of the secret that contains the registry login password. Not used if identity is specified.')
+  passwordSecretRef: string?
+
+  @description('Required. Container registry server.')
+  server: string
+
+  @description('Required. Container registry username.')
+  username: string
+}
+
+@description('Optional. Managed Identity settings for the session pool.')
+type managedIdentitySettingType = {
+  @description('Required. The resource ID of a user-assigned managed identity that is assigned to the Session Pool, or "system" for system-assigned identity.')
+  identity: string
+
+  @description('Required. Use to select the lifecycle stages of a Session Pool during which the Managed Identity should be available. Valid values: "All", "Init", "Main", "None".')
+  lifecycle: string
+}[]?
```

Udt spec 1/2
Types should be nonnullable and singular. Ref spec https://azure.github.io/Azure-Verified-Modules/specs/bcp/res/

To comply to that, you can remove `[]?` from here and move it in the parameter declaration
```suggestion
}
```